src/go/rpk/pkg/cli/cluster/partitions/toggle.go

```diff
@@ -0,0 +1,243 @@
+				cmd.Help()
+				os.Exit(1)
+			}
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+			out.CheckExitCloudAdmin(p)
+
+			cl, err := adminapi.NewClient(fs, p)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			err = runToggle(cmd.Context(), cl, all, topicArg, partitions, "disable")
+			out.MaybeDieErr(err)
+			if all {
+				fmt.Printf("Successfully disabled all partitions in topic %v\n", topicArg[0])
+			} else {
+				fmt.Printf("Successfully disabled the partitions %v\n", partitions)
+			}
+		},
+	}
+	cmd.Flags().StringArrayVarP(&partitions, "partitions", "p", nil, "Comma-separated list of partitions you want to disable. Use --help for additional information")
+	cmd.Flags().BoolVarP(&all, "all", "a", false, "If true, disable all partitions")
+
+	cmd.MarkFlagsMutuallyExclusive("partitions", "all")
+	return cmd
+}
+
+func runToggle(ctx context.Context, cl *adminapi.AdminAPI, all bool, topicArg, partitionFlag []string, verb string) error {
+	isDisable := verb == "disable"
+	if all {
+		ns, topicName := nsTopic(topicArg[0])
+		err := cl.ToggleAllTopicPartitions(ctx, isDisable, ns, topicName)
+		if err != nil {
+			return fmt.Errorf("failed to %v all partitions in topic %v/%v: %v", verb, ns, topicName, err)
+		}
+		return nil
+	}
+	g, egCtx := errgroup.WithContext(ctx)
+	for _, ntp := range partitionFlag {
+		ns, topicName, partitions, err := parsePartition(ntp)
+		if err != nil {
+			return err
+		}
+		if topicName != "" && len(topicArg) > 0 && topicArg[0] != "" {
+			return fmt.Errorf("unable to run the command: topic cannot be specified both as an argument and in the --partition flag")
+		}
+		if topicName == "" {
+			if len(topicArg) == 0 {
+				return fmt.Errorf("unable to run the command: you must specify the topic either as an argument or in the --partition flag")
+			}
+			topicName = topicArg[0]
+		}
+		for _, p := range partitions {
+			p := p
+			g.Go(func() error {
+				err := cl.ToggleTopicPartitions(egCtx, isDisable, ns, topicName, p)
+				if err != nil {
+					return fmt.Errorf("failed to %v partition %v in topic %v/%v: %v", verb, p, ns, topicName, err)
+				}
+				return nil
+			})
+		}
+	}
+	return g.Wait()
+}
+
+var (
+	partitionRe     *regexp.Regexp
+	partitionReOnce sync.Once
+)
+
+// parsePartition parses the partition flag with the format:
+// {namespace}/{topic}/[partitions...]
+// where namespace and topic are optionals, and partitions are comma-separated
+// partitions ID. If namespace is not provided, the function assumes 'kafka'.
+func parsePartition(ntp string) (ns, topic string, partitions []int, rerr error) {
+	partitionReOnce.Do(func() {
+		// Matches {namespace}/{topic}/[partitions...]
+		// - Index 0: Full Match.
+		// - Index 1: Namespace, if present.
+		// - Index 2: Topic, if present.
+		// - Index 3: Comma-separated partitions.
+		partitionRe = regexp.MustCompile(`^(?:(?:([\w-]+)/)?([\w-]+)/)?(\d+(?:,\d+)*)$`)
```

Comment:
I _don't think_ `\w-` matches a Kafka topic because it doesn't match dots. But, easier would be `[^/]+`, which will match up to the next slash (and not really be any more stringent past that).
Otherwise this regex lgtm
So, full change suggestion is:
```
^(?:(?:([^/]+)/)?([^/]+)/)?(\d+(?:,\d+)*)$
```

Reply:
Very good catch, thanks!